src/Filter/AbstractDateFilter.php

```diff
@@ -75,7 +75,22 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
             $endDateParameterName = $this->getNewParameterName($queryBuilder);
 
             if (DateRangeType::TYPE_NOT_BETWEEN === $data['type']) {
-                $this->applyWhere($queryBuilder, sprintf('%s.%s < :%s OR %s.%s > :%s', $alias, $field, $startDateParameterName, $alias, $field, $endDateParameterName));
+                if ($data['value']['start'] || $data['value']['end']) {
+                    $filterStartParameter = $filterEndParameter = null;
+
+                    if ($data['value']['start']) {
+                        $filterStartParameter = sprintf('%s.%s < :%s', $alias, $field, $startDateParameterName);
+                    }
+
+                    if ($data['value']['end']) {
+                        $filterEndParameter = sprintf('%s.%s > :%s', $alias, $field, $endDateParameterName);
+                    }
+
+                    $queryBuilder->orWhere(
+                        $filterStartParameter,
+                        $filterEndParameter
+                    );
```

Comment:
`null` seems to be properly handled: 👍 https://github.com/doctrine/orm/blob/6dc46e78ccf4fe75c80884e7abe41b811f2204b6/lib/Doctrine/ORM/Query/Expr/Base.php#L66